### PR TITLE
ci: Check whether the YBT amount after redeem will be less or equal to YBT amount that is deposited

### PR DESCRIPTION
**Pull request by StefanVitor**

### CODE REVIEW

File: contracts/echidna/EchidnaTempusPool.sol [hunk 1]
```diff
@@ -100,6 +109,32 @@ abstract contract EchidnaTempusPool {
...
+    ) internal {
+        uint256 currentPrincipalAmount = IERC20(address(tempusPool.principalShare())).balanceOf(
+            convertAddressToLimitedSet(fromRecipient)
+        );
+        uint256 currentYieldAmount = IERC20(address(tempusPool.yieldShare())).balanceOf(
+            convertAddressToLimitedSet(fromRecipient)
+        );
+
+        if (currentPrincipalAmount == 0 && currentYieldAmount == 0) {
+            _depositYieldBearing(yieldTokenAmount, fromRecipient);
```
Comment (mijovic) [inline contracts/echidna/EchidnaTempusPool.sol:125]: this should return amount of minted shares, which can be used later for withdrawal

Comment (mijovic) [reply contracts/echidna/EchidnaTempusPool.sol:125]: Instead of having sharesAmount as parameter

Comment (StefanVitor) [reply contracts/echidna/EchidnaTempusPool.sol:125]: But if you think that minted amount is enough for the check, I'll change it to that.

Comment (mijovic) [reply contracts/echidna/EchidnaTempusPool.sol:125]: Also, this will eliminate the condition for principal and yield amount to be 0

**Review by mijovic — APPROVED** [review]

**Pull request merged**